spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/JettyWebSocketHandlerAdapter.java

```diff
@@ -71,65 +63,63 @@ public JettyWebSocketHandlerAdapter(WebSocketHandler handler,
+	@Override
 	public void onWebSocketOpen(Session session) {
-		this.delegateSession = this.sessionFactory.apply(session);
+		this.delegateSession = Objects.requireNonNull(this.sessionFactory.apply(session));
 		this.delegateHandler.handle(this.delegateSession)
-				.checkpoint(session.getUpgradeRequest().getRequestURI() + " [JettyWebSocketHandlerAdapter]")
-				.subscribe(this.delegateSession);
+				.subscribe(new Subscriber<>() {
+					@Override
+					public void onSubscribe(Subscription s) {
+						s.request(Long.MAX_VALUE);
+					}
+
+					@Override
+					public void onNext(Void unused) {
+					}
+
+					@Override
+					public void onError(Throwable t) {
+						delegateSession.onHandlerError(t);
+					}
+
+					@Override
+					public void onComplete() {
+						delegateSession.onHandleComplete();
+					}
+				});
 	}
 
-	@OnWebSocketMessage
+	@Override
 	public void onWebSocketText(String message) {
-		if (this.delegateSession != null) {
-			byte[] bytes = message.getBytes(StandardCharsets.UTF_8);
-			DataBuffer buffer = this.delegateSession.bufferFactory().wrap(bytes);
-			WebSocketMessage webSocketMessage = new WebSocketMessage(Type.TEXT, buffer);
-			this.delegateSession.handleMessage(webSocketMessage.getType(), webSocketMessage);
-		}
+		byte[] bytes = message.getBytes(StandardCharsets.UTF_8);
+		DataBuffer buffer = this.delegateSession.bufferFactory().wrap(bytes);
+		WebSocketMessage webSocketMessage = new WebSocketMessage(Type.TEXT, buffer);
+		this.delegateSession.handleMessage(webSocketMessage);
 	}
 
-	@OnWebSocketMessage
+	@Override
 	public void onWebSocketBinary(ByteBuffer byteBuffer, Callback callback) {
-		if (this.delegateSession != null) {
-			DataBuffer buffer = this.delegateSession.bufferFactory().wrap(byteBuffer);
-			buffer = new JettyDataBuffer(buffer, callback);
-			WebSocketMessage webSocketMessage = new WebSocketMessage(Type.BINARY, buffer);
-			this.delegateSession.handleMessage(webSocketMessage.getType(), webSocketMessage);
-		}
-		else {
-			callback.succeed();
-		}
+		DataBuffer buffer = this.delegateSession.bufferFactory().wrap(byteBuffer);
+		buffer = new JettyDataBuffer(buffer, callback);
+		WebSocketMessage webSocketMessage = new WebSocketMessage(Type.BINARY, buffer);
+		this.delegateSession.handleMessage(webSocketMessage);
 	}
 
-	@OnWebSocketFrame
-	public void onWebSocketFrame(Frame frame, Callback callback) {
-		if (this.delegateSession != null) {
-			if (OpCode.PONG == frame.getOpCode()) {
-				ByteBuffer byteBuffer = (frame.getPayload() != null ? frame.getPayload() : EMPTY_PAYLOAD);
-				DataBuffer buffer = this.delegateSession.bufferFactory().wrap(byteBuffer);
-				buffer = new JettyDataBuffer(buffer, callback);
-				WebSocketMessage webSocketMessage = new WebSocketMessage(Type.PONG, buffer);
-				this.delegateSession.handleMessage(webSocketMessage.getType(), webSocketMessage);
-				return;
-			}
-		}
-
-		callback.succeed();
+	@Override
+	public void onWebSocketPong(ByteBuffer payload) {
```

Comment:
Is there the override for `onWebSocketPing()`?
What about partial?

Reply:
They don't have any logic for WebSocket PING so no point in overriding it.

 They don't handle partial messages either, they only handle complete messages.